### PR TITLE
Finalize v0.2

### PR DESCRIPTION
**Pull request by martinkersner**
# Description

This PR integrates recommendations from mythril and slither. The rest of the findings are not too relevant or caused by incorrect detection.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.

## Summary by CodeRabbit

- **Documentation**
	- Updated deployment script command in the README for clarity.
	- Added new README files with instructions for using Mythril and Slither tools for security analysis.

- **New Features**
	- Introduced a new security analysis file, highlighting potential risks and providing improvement suggestions for smart contracts.

- **Refactor**
	- Updated Solidity pragma versions across multiple contracts and test files for consistency.
	- Simplified function return types in several contracts to improve code clarity and efficiency.

- **Bug Fixes**
	- Adjusted contract logic to handle data submissions more effectively based on freshness, enhancing security and performance.

### CODE REVIEW

**Review by nick-bisonai — APPROVED** [review]
lgtm! just left few questions.
BTW, is there a reason for why all the migration files are being removed?

File: contracts/v0.2/foundry.toml [hunk 1]
```diff
@@ -4,10 +4,9 @@ out = "out"
 libs = ["lib"]
 broadcast="deployments"
 fs_permissions = [{ access = "read-write", path = "./"}]
-memory_limit = 67108864
-solc-version = "0.8.24"
-# optimizer = true
-# optimizer-runs = 10_000_000
+solc-version = "0.8.20"
```
[the remark this answers is not in the document]
Comment (martinkersner) [reply contracts/v0.2/foundry.toml:7]: It supports, but the highest version of published OpenZeppelin is 0.8.20. The tools recommended to use the same version for all.

**Comment by martinkersner** [comment]
> lgtm! just left a question. btw, is there a reason for why all the migration files are being removed?

Migration files are removed because we have not deployed anything officially and they include old format of migration configuration anyway.

**Comment by nick-bisonai** [comment]
and another small question, why does it recommend to remove param names for return values?

**Comment by martinkersner** [comment]
> and another small question, why does it recommend to remove param names for return values?

It recommends it, but it actually does not detect I have removed it. The reason behind that is in my opinion unnecessary variable definition since named variables in `returns` statement are allocated. They are probably removed when solidity optimizer is run, but still they have no purpose to be there.

**Comment by nick-bisonai** [comment]
Okie, thank you for kind explanations!

**Pull request merged**